### ISSUE TITLE
`XlaRuntimeError: UNIMPLEMENTED` when runninq `diffeqsolve` on TPU with x64 enabled

### ISSUE THREAD

**Issue by stefanocortinovis** [patrick-kidger/diffrax#366]
Hi Patrick. This might be quite the edge case, but if I run
```python
from jax import config
config.update("jax_enable_x64", True)

from diffrax import diffeqsolve, Dopri5, ODETerm, SaveAt, PIDController

vector_field = lambda t, y, args: -y
term = ODETerm(vector_field)
solver = Dopri5()
saveat = SaveAt(ts=[0., 1., 2., 3.])
stepsize_controller = PIDController(rtol=1e-5, atol=1e-5)

sol = diffeqsolve(term, solver, t0=0, t1=3, dt0=0.1, y0=1, saveat=saveat,
                  stepsize_controller=stepsize_controller)
```
on TPU with x64 enabled, the following error is raised:
```
XlaRuntimeError: UNIMPLEMENTED: While rewriting computation to not contain X64 element types, XLA encountered an HLO for which this rewriting is not implemented: %bitcast-convert.577 = s64[] bitcast-convert(f64[] %constant.495), metadata={op_name="jit(diffeqsolve)/jit(main)/outer-loop/checkpointed-no-vjp/while/body/nextafter" source_file="ENV/lib/python3.10/site-packages/equinox/internal/_nextafter.py" source_line=9}
```
The snippet runs without issues on CPU/GPU. Any idea what might be going on here?

**Comment by patrick-kidger** [comment]
I believe TPUs don't support float64 natively, and have to do so via software emulation. I'm not sure how that works under the hood, but it might be loosely something like "use two float32s".

Anyway, it seems like XLA:TPU doesn't know how to rewrite a bitcast operation on float64s.

Really this is just an unimplemented feature of the XLA:TPU compiler, but we might be able to avoid it ever being necessary. What happens if you replace the `jnp.where` on this line:

https://github.com/patrick-kidger/diffrax/blob/8aafefcd081d51240ac865662a1767ca999d2455/diffrax/_step_size_controller/adaptive.py#L607

with a `static_select`:

https://github.com/patrick-kidger/diffrax/blob/8aafefcd081d51240ac865662a1767ca999d2455/diffrax/_misc.py#L144

?

**Comment by stefanocortinovis** [comment]
Good catch. Moving from `jnp.where` to `static_select` does indeed solve the issue. I can create a PR right away if you'd like.

**Comment by patrick-kidger** [comment]
Yup, that'd be great!

**Comment by stefanocortinovis** [comment]
Hi again @patrick-kidger, sorry for bringing your attention here again! After our discussion above, I looked a little more in-depth into `nextafter` defined in `equinox.internal`, i.e.
```python
def nextafter(x: Array) -> Array:
    y = jnp.nextafter(x, jnp.inf).
    return jnp.where(x == 0, jnp.finfo(x.dtype).tiny, y)
```

Would it make sense to refactor it to
```python
def nextafter(x: Array) -> Array:
    return x + jnp.finfo(x.dtype).smallest_normal  
```
or am I missing something here? I assume this would have the benefit of ruling out the possibility of the `XlaRuntimeError: UNIMPLEMENTED` error on TPU we discussed above? Same goes for `prevbefore` defined in the same file.
